libraries/tools/kotlin-maven-plugin/src/main/java/org/jetbrains/kotlin/maven/KotlinCompileMojoBase.java

```diff
@@ -133,16 +137,32 @@ public void execute() throws MojoExecutionException, MojoFailureException {
         configureCompilerArguments(arguments, compiler);
         printCompilerArgumentsIfDebugEnabled(arguments, compiler);
 
-        MessageCollector messageCollector = new MavenPluginLogMessageCollector(getLog());
+        MavenPluginLogMessageCollector messageCollector = new MavenPluginLogMessageCollector(getLog());
 
         ExitCode exitCode = compiler.exec(messageCollector, Services.EMPTY, arguments);
 
-        switch (exitCode) {
-            case COMPILATION_ERROR:
-                throw new MojoExecutionException("Compilation error. See log for more details");
-            case INTERNAL_ERROR:
-                throw new MojoExecutionException("Internal compiler error. See log for more details");
-            default:
+        if (exitCode != ExitCode.OK) {
+            throw new KotlinCompilationFailureException(
+                    CollectionsKt.map(messageCollector.getCollectedErrors(), new Function1<Pair<CompilerMessageLocation, String>, CompilerMessage>() {
+                        @Override
+                        public CompilerMessage invoke(Pair<CompilerMessageLocation, String> pair) {
+                            CompilerMessageLocation location = pair.getFirst();
+                            String message = pair.getSecond();
+                            String lineContent = location.getLineContent();
+                            int lineContentLength = lineContent == null ? 0 : lineContent.length();
+
+                            return new CompilerMessage(
+                                    location.getPath(),
+                                    CompilerMessage.Kind.ERROR,
```

Comment:
yes, but in this case it looks not so clear that there is -1 and 0 with special meaning